csharp/src/Client/SchemaConverter.cs

```diff
@@ -58,15 +61,20 @@ public static DataTable ConvertArrowSchema(Schema schema, AdbcStatement adbcStat
                 row[SchemaTableColumn.ColumnName] = f.Name;
                 row[SchemaTableColumn.ColumnOrdinal] = columnOrdinal;
                 row[SchemaTableColumn.AllowDBNull] = f.IsNullable;
-                row[SchemaTableColumn.ProviderType] = f.DataType;
+                row[ArrowTypeName] = f.DataType;
                 Type t = ConvertArrowType(f);
 
                 row[SchemaTableColumn.DataType] = t;
 
+                if (f.HasMetadata && f.Metadata.ContainsKey("logicalType"))
```

Comment:
In general, this is a nice introduction. However, `logicalType` is specific to Snowflake. How can this be generic for different drivers?

Reply:
Can each driver use a distinct schemaConverter? The common schemaConverter.cs preserves all metadata and each driver schemaConverter could based on their requirement and do some customization.

Reply:
I dont think that would work very well when doing interop to other languages. I was thinking of just making the key name configurable to check the metadata that is provided by the provider via the passed value.

Reply:
can the SchemaConverter know what's the current driver?  And we could maintain a dictionary to store driver specific metadata

Reply:
I would rather see the caller of the client specify what the value should be (because the caller knows the driver it is working with) rather than try to maintain that internal to the client library.

Reply:
In general, the other implementations have the driver convert the database type to the proper Arrow type up front (e.g. Snowflake numerics get turned into Arrow decimals), then have a single consistent mapping for Arrow type to language-native type. That way we don't have to worry about multiple conversion targets in each driver, or asking clients to know what driver they are using (isn't ADO.NET meant to abstract some of those details?)